account_statement_completion_compassion/model/statement.py

```diff
@@ -0,0 +1,257 @@
+
+from openerp import api, exceptions, fields, models, _
+from openerp.tools import mod10r
+
+# Name of gifts products
+GIFT_NAMES = ["Birthday Gift", "General Gift", "Family Gift", "Project Gift",
+              "Graduation Gift"]
+
+# Name of gift category
+GIFT_CATEGORY = "Sponsor gifts"
+
+# Name of sponsorship category
+SPONSORSHIP_CATEGORY = "Sponsorship"
+
+import time
+
+
+class AccountStatement(models.Model):
+
+    """ Adds a relation to a recurring invoicer. """
+
+    _inherit = 'account.bank.statement'
+
+    recurring_invoicer_id = fields.Many2one(
+        'recurring.invoicer', 'Invoicer')
+    generated_invoices_count = fields.Integer('Invoices',
+                                              compute='_count_invoices')
+
+    @api.one
+    @api.depends('recurring_invoicer_id')
+    def _count_invoices(self):
+        self.generated_invoices_count = len(
+            self.recurring_invoicer_id.invoice_ids)
+
+    @api.multi
+    def to_invoices(self):
+        self.ensure_one()
+        return {
+            'name': 'Generated Invoices',
+            'view_mode': 'tree,form',
+            'view_type': 'form',
+            'res_model': 'account.invoice',
+            'domain': [('recurring_invoicer_id', '=',
+                        self.recurring_invoicer_id.id)],
+            'type': 'ir.actions.act_window',
+            'target': 'current',
+            'context': {'form_view_ref': 'account.invoice_form',
+                        'journal_type': 'sale'},
+        }
+
+    @api.one
+    def button_invoices(self):
+        invoicer_id = self.recurring_invoicer_id.id
+
+        return {
+            'name': 'Generated Invoices',
+            'view_mode': 'tree,form',
+            'view_type': 'form',
+            'res_model': 'account.invoice',
+            'domain': [('recurring_invoicer_id', '=', invoicer_id)],
+            'type': 'ir.actions.act_window',
+            'target': 'current',
+            'context': {'form_view_ref': 'account.invoice_form',
+                        'journal_type': 'sale'},
+        }
+
+    @api.multi
+    def button_confirm_bank(self):
+        """Confirm invoices generated by statment lines."""
+        res = super(AccountStatement, self).button_confirm_bank()
+        for statement in self:
+            invoices = statement.line_ids.mapped('invoice_id')
+            invoices.signal_workflow('invoice_open')
+        return res
+
+    @api.multi
+    def button_cancel(self):
+        """ Cancel invoices generated by statement lines. """
+        res = super(AccountStatement, self).button_cancel()
+        invoices = self.mapped('line_ids.invoice_id')
+        invoices.action_cancel_draft()
+        return res
+
+
+class AccountStatementLine(models.Model):
+    """ Adds products to statement lines to generate invoices. """
+
+    _inherit = 'account.bank.statement.line'
+
+    product_id = fields.Many2one('product.product', 'Product')
```

Comment:
Remove this field. We use that in reconcile view and don't need it on statement lines.